api4/channel.go

```diff
@@ -1513,6 +1549,27 @@ func addChannelMember(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	categoryId, categoryIdOK := props["category_id"].(string)
+	if categoryIdOK && categoryId != "" && !model.IsValidId(categoryId) {
+		c.SetInvalidParam("category_id")
+		return
+	}
+
+	var category *model.SidebarCategoryWithChannels
+	if categoryIdOK && len(categoryId) == 26 {
+		if !c.App.SessionHasPermissionToCategory(*c.AppContext.Session(), member.UserId, channel.TeamId, categoryId) {
+			c.SetInvalidParam("category")
+			return
+		}
+		categoryData, categoryDataErr := c.App.GetSidebarCategory(categoryId)
```

Comment:
Nit: just `err` is fine.
- If you run into shadowing issues, use `err2`.
- If you need to distinguish between `AppError` and `error` use `appErr`.